library/include/internal/hipsolver-refactor.h

```diff
@@ -0,0 +1,215 @@
+/* ************************************************************************
+ * Copyright (C) 2023 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#ifndef HIPSOLVER_REFACTOR_H
+#define HIPSOLVER_REFACTOR_H
+
+#include "hipsolver-types.h"
+
+typedef void* hipsolverRfHandle_t;
```

Comment:
We could maybe make this an opaque struct pointer, but this is consistent with our current handle. I think we want to _consider_ making this type-safe in ROCm 6.0, but I actually have a bunch of open questions about ABI compatibility between the rocm and cuda variants of HIP and the HIP math libraries that I want to get answered before making a decision there.

Reply:
Yeah, my plan was to keep things consistent with the existing `typedefs` for now, and then transition to opaque pointers in the next major version release.